addons/common/init_perFrameHandler.sqf

```diff
@@ -16,22 +16,36 @@ GVAR(nextFrameBufferA) = [[[], {GVAR(nextFrameNo) = diag_frameno;}]];
 GVAR(nextFrameBufferB) = [];
 GVAR(waitUntilAndExecArray) = [];
 
+GVAR(deletedPFHIndices) = [];
 // per frame handler system
 [QFUNC(onFrame), {
     private _tickTime = diag_tickTime;
     call FUNC(missionTimePFH);
 
     // Execute per frame handlers
     {
-        _x params ["_function", "_delay", "_delta", "", "_args", "_handle"];
+        _x params ["_function", "_delay", "_delta", "", "_args", "_handle", "_isActiv"];
 
-        if (diag_tickTime > _delta) then {
+        if (diag_tickTime > _delta && _isActiv) then {
             _x set [2, _delta + _delay];
             [_args, _handle] call _function;
             false
         };
     } count GVAR(perFrameHandlerArray);
 
+    if !(GVAR(deletedPFHIndices) isEqualTo []) then {
+        {
+            GVAR(perFrameHandlerArray) set [_x, objNull];
+        } count GVAR(deletedPFHIndices);
+
+        GVAR(perFrameHandlerArray) = GVAR(perFrameHandlerArray) - [objNull];
+
+        {
+            _x params ["", "", "", "", "", "_handle"];
+            GVAR(PFHhandles) set [_handle, _forEachIndex];
```

Comment:
Wait.. You are resetting the handles here? meaning everyone that saved his handles will delete some random PFH but not the one they intended to delete.